CMakeLists.txt

```diff
@@ -33,21 +33,47 @@ add_definitions(-DTEST_DATA_DIR="${PROJECT_SOURCE_DIR}/testdata")
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/external/cmake-modules)
 
-include(lint)
 include(CTest)
 
-# External projects
-if (BUILD_TESTING)
-  #add_subdirectory(external/gtest-1.8.1 googletest EXCLUDE_FROM_ALL)
-  # installed as git submodule - if this is your first clone you need to
-  # git submodule init
-  # git submodule update
-  # This is a cmake module and needs no further input from you
+# Gather External Projects
+find_package(Git)
+if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+  # Update submodules as needed, see https://cliutils.gitlab.io/modern-cmake/chapters/projects/submodule.html
+  option(GIT_SUBMODULE "Check submodules during build" ON)
+  if(GIT_SUBMODULE)
+      message(STATUS "Submodule update")
+      execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
```

Comment:
The `--recursive` flag seems unnecessary at the moment -- it triggers the cloning and checkout of the googletest submodules of *our* submodules (i.e., `external/CLI11/extern/googletest` and `external/rapidjson-1.1.0/thirdparty/gtest`), both of which are not really required for building our project. Having them there doesn't add anything to our compilation, but I'd be inclined to avoid unnecessarily checking them out.

Reply:
I suppose this may want to be avoided if storage space is an issue in some enivironment. I don't mind too much either way.